### PR TITLE
Update vs-2017.3.* folders and files to be ide.*

### PR DESCRIPTION
**Pull request by phenning**
**PR Title**
Updates the templates to use ide.host.json and associated folder name. 

**PR Description**
This change updates the templates to use the preferred fallback host name instead of the old vs-2017.3 host name. This also removes some metadata which is no longer used by the new project creation experience.

Note that we left the metadata around authentications here, even though it may not be strictly needed. This may be able to be streamlined and removed in vNext.

Addresses #34134

### CODE REVIEW

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/.template.config/ide.host.json [hunk 1]
```diff
@@ -1,22 +1,7 @@
 {
   "$schema": "http://json.schemastore.org/vs-2017.3.host",
-  "name": {
-    "text": "Blazor Server App",
```
Comment (mkArtakMSFT) [inline src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/.template.config/ide.host.json:4]: Where from this name is going to be read going forward?

Comment (phenning) [reply src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/.template.config/ide.host.json:4]: All the strings come from the strings.json currently https://github.com/dotnet/aspnetcore/blob/5aebd9b472c70edc38b897006b119f06ed29c0dd/src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/.template.config/en/strings.json#L4

In the future, there will be an alternative way to localize the name and descriptions of the templates. Some of that work is done in the NET 6 template engine, but we aren't pushing to move to it until vNext

**Review by mkArtakMSFT — APPROVED** [review]
Thanks @phenning!

**Comment by phenning** [comment]
Is this OK to merge?

**Pull request merged**